src/contexts/TitleContext.tsx

```diff
@@ -0,0 +1,218 @@
+		"When an unexpected enemy emerges and threatens global safety and security, Nick Fury, director of the international peacekeeping agency known as S.H.I.E.L.D., finds himself in need of a team to pull the world back from the brink of disaster. Spanning the globe, a daring recruitment effort begins!",
+	tmdbId: 24428,
+};
+
+const dummyFilms: { [id: number]: PreviewTitleDto } = {
+	1: {
+		id: 1,
+		name: "The Incredible Hulk",
+		type: "movie",
+		smallPosterUrl:
+			"https://www.themoviedb.org/t/p/w600_and_h900_bestv2/gKzYx79y0AQTL4UAk1cBQJ3nvrm.jpg",
+		releasedAtUtc: new Date("2008-06-12"),
+	},
+	2: {
+		id: 2,
+		name: "Iron Man 2",
+		type: "movie",
+		smallPosterUrl:
+			"https://www.themoviedb.org/t/p/w600_and_h900_bestv2/6WBeq4fCfn7AN0o21W9qNcRF2l9.jpg",
+		releasedAtUtc: new Date("2010-05-07"),
+	},
+	3: {
+		id: 3,
+		name: "Thor",
+		type: "movie",
+		smallPosterUrl:
+			"https://www.themoviedb.org/t/p/w600_and_h900_bestv2/prSfAi1xGrhLQNxVSUFh61xQ4Qy.jpg",
+		releasedAtUtc: new Date("2011-05-06"),
+	},
+	4: {
+		id: 4,
+		name: "Captain America: The First Avenger",
+		type: "movie",
+		smallPosterUrl:
+			"https://www.themoviedb.org/t/p/w600_and_h900_bestv2/vSNxAJTlD0r02V9sPYpOjqDZXUK.jpg",
+		releasedAtUtc: new Date("2011-07-22"),
+	},
+	5: {
+		id: 5,
+		name: "Iron Man 3",
+		type: "movie",
+		smallPosterUrl:
+			"https://www.themoviedb.org/t/p/w600_and_h900_bestv2/qhPtAc1TKbMPqNvcdXSOn9Bn7hZ.jpg",
+		releasedAtUtc: new Date("2013-04-18"),
+	},
+};
+
+const dummyRelations: {
+	title: PreviewTitleDto;
+	relevance: RelationRelevance;
+}[] = [
+	{
+		title: dummyFilms[1],
+		relevance: "could",
+	},
+	{
+		title: dummyFilms[2],
+		relevance: "should",
+	},
+	{
+		title: dummyFilms[3],
+		relevance: "should",
+	},
+	{
+		title: dummyFilms[4],
+		relevance: "should",
+	},
+];
+
+const dummySequences: {
+	[orderId: number]: Sequence;
+} = {
+	[-1]: {
+		previous: dummyFilms[4],
+		next: dummyFilms[5],
+	},
+};
+
+interface TitleContextType {
+	title: CommonTitleDto;
+	setTitle: (titleId: TitleDto["id"]) => void;
+	getRelations: () => Promise<DependencyOrderTitleDto>;
+	getSequences: (orderId: number) => Promise<SequentialOrderTitleDto>;
+}
+
+export const TitleContext = createContext<TitleContextType>({
+	title: {
+		id: -1,
+		name: "",
+		type: "movie",
+	},
+	setTitle: () => {},
+	getRelations: async () => ({
+		id: -1,
+		name: "",
+		type: "movie",
+		order: "relational",
+		relations: [],
+	}),
+	getSequences: async () => ({
+		id: -1,
+		name: "",
+		type: "movie",
+		order: "sequential",
+		orderId: -1,
+		previous: undefined,
+		next: undefined,
+	}),
+});
+
+export const TitleProvider = ({ children }: { children: React.ReactNode }) => {
+	const [title, setTitle] = useState<CommonTitleDto>({
+		id: -1,
+		name: "",
+		type: "movie",
+	});
+
+	const [relations, setRelations] = useState<Relation[] | null>(null);
+
+	const [sequences, setSequences] = useState<{
+		// -1 = release order
+		[orderId: number]: Sequence;
+	}>({});
```

Comment:
You can simplify this type slightly :)
```suggestion
    // An orderId of -1 is reserved for the release order
	const [sequences, setSequences] = useState<Record<number, Sequence>>({});
```